Libraries/src/Amazon.Lambda.Annotations/README.md

```diff
@@ -68,37 +187,141 @@ public class Functions
 }
 ```
 
-and place in your template:
+```json
+    "CloudCalculatorFunctionsAddGenerated": {
+      "Type": "AWS::Serverless::Function",
+      "Metadata": {
+        "Tool": "Amazon.Lambda.Annotations",
+        "SyncedEvents": [
+          "RootGet"
+        ]
+      },
+      "Properties": {
+        "Runtime": "dotnet6",
+
+...
+        "Role": {
+          "Fn::GetAtt": [
+            "LambdaRoleParameter",
+            "Arn"
+          ]
+        }
+      }
+    },
+```
+
+## Amazon API Gateway example
+
+This example creates a REST API through Amazon API Gateway that exposes the common arithmetic operations. 
+
+To avoid putting business logic inside the REST API a separate calculator service is created to encapsulated the logic of the arithmetic operations. Here is both the 
+calculator service's interface and default implementation.
+
+```csharp
+public interface ICalculatorService
+{
+    int Add(int x, int y);
+
+    int Substract(int x, int y);
+
+    int Multiply(int x, int y);
+
+    int Divide(int x, int y);
+}
+
+public class DefaultCalculatorService : ICalculatorService
+{
+    public int Add(int x, int y) => x + y;
+
+    public int Substract(int x, int y) => x - y;
+
+    public int Multiply(int x, int y) => x * y;
+
+    public int Divide(int x, int y) => x / y;
+}
+```
+
+The startup class contains the `LambdaStartup` attribute identifying it as the class to configure the services registered in the dependency injection framework. 
+Here the `ICalculatorService` is registered as a singleton service in the collection of services.
 
+```csharp
+[LambdaStartup]
+public class Startup
+{
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddSingleton<ICalculatorService, DefaultCalculatorService>();
+    }
+}
 ```
 
-  "Parameters": {
-    "LambdaExecutionRole": {
-      "Type": "String"
+Since the `ICalculatorService` is registered as a singleton the service is injected into the Lambda function via the constructor. 
+If the registered service is registered as scoped or transient and a new instance is need for each Lambda invocation then the
```

Comment:
If the registered service is registered as scoped or transient, and a new instance is needed for each Lambda invocation, then the
`FromServices` attribute should be used on a method parameter of the Lambda function. 

*Typo fixed

Reply:
Fixed